### PR TITLE
Enable INC for llava models and change softmax to use torch.nn.functional.softmax as its supported module by INC

### PR DESCRIPTION
**Pull request by tthakkal**
# What does this PR do?

Enable INC for llava models and change softmax to use torch.nn.functional.softmax  as its supported module by INC.
https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-functions

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/image-to-text/run_pipeline.py [hunk 1]
```diff
@@ -36,6 +36,44 @@
...
+
+        config = FP8Config.from_json_file(args.quant_config)
+        if config.measure:
+            model = prepare(model, config)
+        elif config.quantize:
+            model = convert(model, config)
+    else:
+        import habana_quantization_toolkit
+
+        habana_quantization_toolkit.prep_model(model)
```
Comment (yafshar) [inline examples/image-to-text/run_pipeline.py:56]:
```suggestion
        from habana_quantization_toolkit import prep_model

        prep_model(model)
```

personal suggestion!

File: examples/image-to-text/run_pipeline.py [hunk 2]
```diff
@@ -36,6 +36,44 @@
...
+        except ImportError:
+            raise ImportError(
+                "Module neural_compressor is missing. Please use a newer Synapse version to use quantization, or set the environment variable to USE_INC=0"
+            )
+
+        finalize_calibration(model)
+    else:
+        import habana_quantization_toolkit
+
+        habana_quantization_toolkit.finish_measurements(model)
```
Comment (yafshar) [inline examples/image-to-text/run_pipeline.py:74]:
```suggestion
        from habana_quantization_toolkit import finish_measurements

        finish_measurements(model)
```

**Comment by yafshar** [comment]
@tthakkal, please run `GAUDI2_CI=1 RUN_SLOW=true python -m pytest tests/transformers/tests/models/ -s -v` before and after changes and make sure that no new issue is introduced.

File: optimum/habana/transformers/models/clip/modeling_clip.py [hunk 3]
```diff
@@ -60,7 +60,7 @@ def __init__(self):
         super().__init__()
 
     def forward(self, x, dim=None, invAttnHead=None):
-        return torch.ops.hpu.softmax_fp8(x, dim, None, None, invAttnHead)
```
Comment (yafshar) [inline optimum/habana/transformers/models/clip/modeling_clip.py:63]: @tthakkal there are other instances of softmax_fp8 in https://github.com/huggingface/optimum-habana/blob/0027e320b3f6c3f8a98d733ae9a42901de8a1e9d/optimum/habana/diffusers/models/attention_processor.py#L42 & https://github.com/huggingface/optimum-habana/blob/0027e320b3f6c3f8a98d733ae9a42901de8a1e9d/optimum/habana/transformers/models/falcon/modeling_falcon.py#L139

probably we can update those here

Comment (tthakkal) [reply optimum/habana/transformers/models/clip/modeling_clip.py:63]: first one is nowhere used, second one runs only for fp8 use case, so not impacted.

File: examples/image-to-text/run_pipeline.py (hunk 2 above)
Comment (tthakkal) [reply examples/image-to-text/run_pipeline.py:74]: This code is copied from text-generation sample, will leave it as is to keep it consistent

File: examples/image-to-text/run_pipeline.py [hunk 4]
```diff
@@ -169,18 +207,14 @@ def main():
         generator.model = wrap_in_hpu_graph(generator.model)
 
     if args.quant_config:
-        import habana_quantization_toolkit
-
-        habana_quantization_toolkit.prep_model(generator.model)
-
-        htcore.hpu_initialize(generator.model)
```
Comment (yafshar) [inline examples/image-to-text/run_pipeline.py:176]: @tthakkal you removed this initialization. Don't we need it anymore?

Comment (tthakkal) [reply examples/image-to-text/run_pipeline.py:176]: Added back the code, interestingly its missing in text-generation sample too. but it works without.

**Review by mandy-li — APPROVED** [review]
LGTM

**Comment by mandy-li** [comment]
@regisss , please review and  merge if ok for you

**Review by regisss — APPROVED** [review]
